opentaxii/server.py

```diff
@@ -85,20 +157,25 @@ def _create_services(self, service_entities):
                 discovery_services.append((service, advertised))
 
         for service, advertised in discovery_services:
-            service.set_advertised_services([
-                s for s in services if s.id in advertised])
+            service.set_advertised_services([s for s in services if s.id in advertised])
 
         return services
 
+    def get_endpoint(self, relative_path: str) -> Callable[[], Response]:
+        """Get first endpoint matching relative_path."""
+        for endpoint in self.get_services():
+            if endpoint.path == relative_path:
+                return functools.partial(self.handle_request, endpoint=endpoint)
```

Comment:
I've improved the typing annotation instead, it's now `Optional`. This makes more sense, since the code that consumes this already takes `None` responses into account. No reason for raising anything.